packages/metadata-common/src/MetadataEditor.tsx

```diff
@@ -100,6 +102,32 @@ export class MetadataEditor extends ReactWidget {
         if (!this.name) {
           this.title.label = `New ${this.schemaDisplayName}`;
         }
+        this.sortedFields = [];
+        const categories: string[] = [];
+        // Find categories of all schema properties
+        for (const schemaProperty in this.schema) {
+          if (!this.schema[schemaProperty].uihints) {
+            this.schema[schemaProperty].uihints = {};
+          }
+          if (!this.schema[schemaProperty].uihints.category) {
+            this.schema[schemaProperty].uihints.category = null;
+          }
+          const category = this.schema[schemaProperty].uihints.category;
+          if (!categories.includes(category)) {
+            categories.push(category);
```

Comment:
ok so this took me some time to understand, but this add a `null` category doesn't it? That why it still grabs fields without a category. I'm not sure I like the idea of having a `null` in the array for this, would an empty string instead leave the same?